gtdynamics/optimizer/EqualityConstraint.h

```diff
@@ -135,7 +135,33 @@ class VectorExpressionEquality : public EqualityConstraint {
 };
 
 /// Container of EqualityConstraint.
-typedef std::vector<EqualityConstraint::shared_ptr> EqualityConstraints;
+// typedef std::vector<EqualityConstraint::shared_ptr> EqualityConstraints;
+
+class EqualityConstraints : public std::vector<EqualityConstraint::shared_ptr> {
```

Comment:
I think you should still use the typedef (but use `using`)

Reply:
done